modules/programs/ssh.nix

```diff
@@ -355,6 +355,16 @@ in
       '';
     };
 
+    addKeysToAgent = mkOption {
+      type = types.enum [ "yes" "no" "ask" "confirm" ];
```

Comment:
Unfortunately I don't think an `enum` would work in this case since it wouldn't allow a value with a timeout, such as "1h" or "confirm 1h". If we add this option then I suspect it would have to be a plain `types.str`.

Reply:
My concerns persist, but I've changed the PR as suggested. I've rebased and ran all tests again. I leave the final decision to @rycee . Feel free to close or merge.